apps/barista-design-system/src/components/ba-decision-graph/ba-decision-graph-node/ba-decision-graph-node.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+} from '@angular/core';
+import { BaUxdNode, BaUxdEdge } from '@dynatrace/shared/barista-definitions';
+import { SafeHtml, DomSanitizer } from '@angular/platform-browser';
+
+@Component({
+  selector: 'ba-decision-graph-node',
+  templateUrl: './ba-decision-graph-node.html',
+  styleUrls: ['./ba-decision-graph-node.scss'],
+})
+export class BaDecisionGraphNode implements OnChanges {
+  @Input()
+  node: BaUxdNode | undefined;
+
+  /** Array of all nodes and edges */
+  @Input()
+  decisionGraphData: BaUxdNode[] = [];
+
+  @Output('startOver')
+  startOver = new EventEmitter<void>();
+
+  /** Array of all nodes and edges which should be displayed */
+  decisionGraphSteps: BaUxdNode[] = [];
+
+  /** @internal Whether the Undo button in template is displayed */
+  _started: boolean = false;
+
+  constructor(private _sanitizer: DomSanitizer) {}
+
+  ngOnChanges(): void {
+    this.resetProgress();
+    this.decisionGraphSteps.push(this.node!);
```

Comment:
This might be a little bit drastic. I guess you want to do that when either `node` or the `decisionGraphData` input change. You could move these two lines into a function and add them to the setter of these two inputs. If there are ever going to be more inputs on the node, you would have to deal with that to not inadvertently reset the graph.

Reply:
In that case, I want to react when the user has changed the startnode, which results in resetting the current progress and setting the newly picked startnode.